flowmachine/flowmachine/features/subscriber/contact_balance.py

```diff
@@ -81,10 +81,13 @@ def __init__(
         self.subscriber_identifier = subscriber_identifier
         self.exclude_self_calls = exclude_self_calls
 
-        if self.direction == "both":
+        if self.direction not in ("both", "in", "out"):
+            raise ValueError("Unidentified direction: {}".format(self.direction))
```

Comment:
Thanks for adding the check for valid values of `direction` back in. Just as a minor suggestion, I'd add an explicit `else` clause below and raise the ValueError in there.

in this particular case the logic is pretty obvious, but we have/had other places in the code where explicit `else` clauses were missing and there is no way to tell whether this is intentional or the logic is flawed (and in some cases it was actually a bug). So I think we should make it a habit to always add an `else` clause - even if it just contains `pass` and a comment of why the "else" case doesn't need to be explicitly treated. This makes it much easier for the reader to verify the logic at a glance.

Again, not really an issue here, just as a general comment for our general coding habits.